### DISCUSSION TITLE
[Android] Got the path, but have no access?

### DISCUSSION THREAD

**Discussion by GiovanniMeloni** [yasirkula/UnityNativeFilePicker#24] [Q&A] [answered]
Hello! 
First I'd like to thank you for developing this free plugin, its amazing seeing very useful tools like these being shared with other developers.

I've followed the quick tutorial in which explains how to use the library in order to open the file picker in Android, and it was successful on that regard. I click on a button, the system asks for access permission, I agree and select an image. But the image doesnt load on my test scene. 
This is how I try to get the image from the path, at least that's how I've seen people accessing files in Android, would that be correct? 
```
using (UnityWebRequest uwr = UnityWebRequestTexture.GetTexture(path)){
            yield return uwr.SendWebRequest();

            if (uwr.result != UnityWebRequest.Result.Success)
            {
                Debug.LogError(uwr.error);
            }
            else
            {
                Texture2D originalTexture = ((DownloadHandlerTexture)uwr.downloadHandler).texture;
```
I'm aware that `NativeFilePicker.PickFile` is asynchronous so I have been using logs to make sure methods and callbacks are called correctly (first I get the path from PickFile and then I try to get said file from the path)

Ive been using Android Studio to get the logs and this is the error I get:

```
2022-05-26 14:17:24.452 15437-15493/com.VenturionVR.IDOMEDLaViSIM E/Unity: Cannot connect to destination host
     #0 0xb8017deb (libunity.so) GetStacktrace(int) 0x2a
     #1 0xb84ad889 (libunity.so) DebugStringToFile(DebugStringToFileData const&) 0x11c
     #2 0xb7d416cd (libunity.so) DebugLogHandler::Internal_Log(LogType, LogOption, core::basic_string<char, core::StringStorageDefault<char> >, Object*) 0x60
     #3 0xb7d41615 (libunity.so) DebugLogHandler_CUSTOM_Internal_Log(LogType, LogOption, ScriptingBackendNativeStringPtrOpaque*, ScriptingBackendNativeObjectPtrOpaque*) 0xe0
     #4 0xb7245b98 (libil2cpp.so) ? 0x0
     #5 0xb5561c7c (libil2cpp.so) ? 0x0
     #6 0xb72550f4 (libil2cpp.so) ? 0x0
     #7 0xb5475a6c (libil2cpp.so) ? 0x0
     #8 0xb54758b8 (libil2cpp.so) ? 0x0
     #9 0xb7f463b1 (libunity.so) scripting_method_invoke(ScriptingMethodPtr, ScriptingObjectPtr, ScriptingArguments&, ScriptingExceptionPtr*, bool) 0x5c
     #10 0xb7f52edf (libunity.so) ScriptingInvocation::Invoke(ScriptingExceptionPtr*, bool) 0x5e
     #11 0xb7f5c63b (libunity.so) Coroutine::InvokeMoveNext(ScriptingExceptionPtr*) 0x66
     #12 0xb7f5c367 (libu
```

Any help or suggetions are welcome

**Comment by GiovanniMeloni** [comment, marked as answer]
I did not know NativeGallery, another plugin from this same creator, was a thing before making this post, NativeGallery.LoadImageAtPath helped me with this problem.

**Reply by yasirkula** [reply]
Happy to hear that! If you're only interested in picking images/videos, than you can get rid of NativeFilePicker and entirely use NativeGallery for that task.

**Comment by Mario-Henrique** [comment]
Hello Giovanni! to solve this, you need put "file:///" to windows or "file://" to android. At this line:

`using (UnityWebRequest uwr = UnityWebRequestTexture.GetTexture("file://" + path)){`

**Reply by sriram-balaganeshan** [reply]
I tried but my android storage is not accessed can you please

**Reply by yasirkula** [reply]
What was the error you received?
